dask/dataframe/groupby.py

```diff
@@ -1032,7 +1032,7 @@ def aggregate(self, arg, split_every, split_out=1):
                    split_out=split_out, split_out_setup=split_out_on_index)
 
     @insert_meta_param_description(pad=12)
-    def apply(self, func, meta=no_default):
+    def apply(self, func, *args, **kwargs):
```

Comment:
Is there any reasonable way to check that `args[0]` is in fact a meta and give deprecation warning? In py3 you can have positionally default argument, that would work too to avoid API breakage.

Reply:
Precisely, this really bugs me.

But also with positially default arguments, I can totally see people do it in this way:

```
df.groupby().apply(f, arg1, arg2)
```

And fail miserably.

Another way is to make meta a required argument.

Reply:
I'm fine with having things fail here, people really should be using keyword arguments by name instead of by position.